spring-batch-infrastructure/src/main/java/org/springframework/batch/item/KeyValueItemWriter.java

```diff
@@ -22,21 +23,20 @@
  *
  * @author David Turanski
  * @author Mahmoud Ben Hassine
+ * @author Stefano Cordio
  * @since 2.2
  *
  */
 public abstract class KeyValueItemWriter<K, V> implements ItemWriter<V>, InitializingBean {
 
-	protected Converter<V, K> itemKeyMapper;
+	protected @Nullable Converter<V, K> itemKeyMapper;
 
 	protected boolean delete;
 
 	@Override
-	public void write(Chunk<? extends V> items) throws Exception {
-		if (items == null) {
-			return;
-		}
-		for (V item : items) {
+	public void write(Chunk<? extends V> chunk) throws Exception {
+		for (V item : chunk) {
+			@SuppressWarnings({ "DataFlowIssue", "NullAway" })
```

Comment:
I find excluding both `DataFlowIssue` and `NullAway` redundant, so I raised uber/NullAway#1230.